demo/admin/src/utils/pageTreeNodeCategoryMapping.ts

```diff
@@ -1,22 +1,12 @@
 import { GQLPageTreeNodeCategory } from "@src/graphql.generated";
+import { kebabCase, pascalCase } from "change-case";
 
 function categoryToUrlParam(category: GQLPageTreeNodeCategory): string {
-    switch (category) {
-        case "TopMenu":
-            return "top-menu";
-        case "MainNavigation":
-        default:
-            return "main-navigation";
-    }
+    return kebabCase(category);
 }
 
 function urlParamToCategory(param: string | undefined): GQLPageTreeNodeCategory | undefined {
-    switch (param) {
-        case "top-menu":
-            return "TopMenu";
-        case "main-navigation":
-            return "MainNavigation";
-    }
+    return param ? (pascalCase(param) as GQLPageTreeNodeCategory) : undefined;
```

Comment:
I thought it would be a convenient simplification but you're right it's actually lying. 

Now I extracted the categories from App.tsx to a dedicated file and use them to validate the category in `urlParamToCategory()`